src/server.c

```diff
@@ -316,7 +327,15 @@ static bool handle_server_work(PgSocket *server, PktHdr *pkt)
 			disconnect_server(server, true, "invalid server parameter");
 			return false;
 		}
-		/* fallthrough */
+		/* ErrorResponse and CommandComplete show end of copy mode */
+		if (server->copy_mode) {
+			server->copy_mode = false;
+
+			/* it's impossible to track sync count over copy */
+			if (client)
+				client->expect_rfq_count = 0;
+		}
```

Comment:
We should do something with the outstanding_requests queue when receiving an ErrorResponse from the server, probably it should clear the complete queue until the next `Sync` message.

Reply:
Why do we need to do something special with outstanding messages?
As far as I understand, extended protocol assumes in case of errors messages can be ignores by server until next Sync message. So why we can not just try to deliver outstanding messages to the server even in some error sis detected?

Reply:
Yeah, I think forwarding whatever is being sent is totally fine. I mainly meant that the server might not send responses for each of the outstanding requests until the next Sync. So we need to remove those outstanding messages from the queue, until we find a Sync.

And now that I think about it more: If we have not received a Sync from the client yet, then we should not add stuff to the queue until we do.